### PR TITLE
Display current args at edit prompt

### LINKED ISSUE #37921: quarkus dev broken for command mode arguments
### Describe the bug

using quarkus dev with command mode seem to have regressed.

do this:

`quarkus create app -x picocli whoops`

then run:
`quarkus dev this is my name`

and it will print out "this is my name" and in bottom line have:

`Press [space] to restart, [e] to edit command line args (currently 'this is my name'), [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>`

now press 'e' to edit and notice that you are shown an empty line. Type in `something else completely` and press enter and you get:

`Unmatched arguments from index 1: 'completely', 'different'
Usage: greeting [-hV] <name>
      <name>      Your name.
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.`

but the line still shows:

`Press [space] to restart, [e] to edit command line args (currently 'this is my name'), [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>`

so multiple issues in the above:

1) `quarkus dev whatever i put` the latter part gets passed in as *one* string ("whatever i put") rather than actual individual arguments. It should be an array of strings ("whatever", "i", "put") rather than one big string. 
   note: `quarkus dev "whatever I put"` should give you one string and not an array of string.
   the edit command actually seem to do this correct. its just the passing of parameter that is wrong
2)  the line at bottom does not show the current command line arg but what was passed in originally
3) the edit does not actually let you edit the command line. just give you blank line. Sometimes characters are not even shown when typing.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by MaciejDromin**
It fixes #37921 - point 3.
At the edit prompt it shows now current command line arguments, as shown in the screenshot below.
edit-demo

### CODE REVIEW

**Comment by gsmet** [comment]
@maxandersen could you test that one?

**Comment by maxandersen** [comment]
I tried this and it is better but still broken.

tried capture it in this recording:


https://github.com/quarkusio/quarkus/assets/54129/ef557e71-1c74-439c-b272-b840f3e622e5

editing shows the current arg line correctly, but when I press enter it still shows "current arg: >" making me believe it is still asking for input.

But it is actually in devmode input mode as shorthand keys like d for debug etc works.

File: core/deployment/src/main/java/io/quarkus/deployment/console/ConsoleStateManager.java [hunk 1]
```diff
@@ -70,7 +70,13 @@ public void accept(int[] ints) {
                     Holder command = commands.get(key);
                     if (command != null) {
                         if (command.consoleCommand.getReadLineHandler() != null) {
-                            QuarkusConsole.INSTANCE.doReadLine();
+                            if (key == 'e') {
+                                QuarkusConsole.INSTANCE.doReadLineWithPrompt("Current args: "
```
Comment (maxandersen) [inline core/deployment/src/main/java/io/quarkus/deployment/console/ConsoleStateManager.java:74]: what do you think about just having it be:

`"state list"> `

instead of:

`Current args: "state list">` 

just to save some characters in width?

otherwise maybe split over two lines and not use `current` as you can't "escape" to reuse it?

`Enter new arguments (was "state list"):
>`

**Review by maxandersen — CHANGES_REQUESTED** [review]
good changes but behavior still broken after pressing enter.

File: core/deployment/src/main/java/io/quarkus/deployment/console/ConsoleStateManager.java (hunk 1 above)
Comment (MaciejDromin) [reply core/deployment/src/main/java/io/quarkus/deployment/console/ConsoleStateManager.java]: Prompt changed

**Comment by MaciejDromin** [comment]
> I tried this and it is better but still broken.
> editing shows the current arg line correctly, but when I press enter it still shows "current arg: >" making me believe it is still asking for input.
> 
> But it is actually in devmode input mode as shorthand keys like d for debug etc works.

I think this might be some weird issue with different terminals, that's why I wasn't encountering this issue from your original issue:
> Sometimes characters are not even shown when typing.

I am not seeing anything like this - maybe it is something related to Your terminal or OS? I checked this both in ZSH and Bash and it is fine, check the recording below:

https://github.com/quarkusio/quarkus/assets/39798354/8e660c71-30a9-4316-8242-62cb1a694617

**Comment by MaciejDromin** [comment]
Hey @maxandersen! Have you maybe had a chance to look at the video I posted? I'm checking on the status of this PR.

**Comment by maxandersen** [comment]
Sorry. Missed the update. I'll check when back at my laptop.

**Comment by gsmet** [comment]
@maxandersen I think we should merge this as it's better than what we had, even if it didn't fix the whole issue for you (apparently, it works for the OP so it might be environment-dependent).

WDYT?

**Comment by maxandersen** [comment]
you are probably right - let me try again.

**Comment by MaciejDromin** [comment]
Just wanted to let you know, that if there are still some issues with this PR which I will be able to reproduce, then I will be happy to fix them

**Comment by maxandersen** [comment]
i've tried to build this pr but keeps having weird issues building. 

recent one is:
```
ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.9.22:test-compile (test-compile) on project arc-tests: Compilation failure: Compilation failure:
[ERROR] /Users/manderse/code/quarkusio/quarkus/independent-projects/arc/tests/src/test/kotlin/io/quarkus/arc/test/interceptors/InterceptableSuspendMethodTest.kt:[3,23] Unresolved reference: Arc
[ERROR] /Users/manderse/code/quarkusio/quarkus/independent-projects/arc/tests/src/test/kotlin/io/quarkus/arc/test/interceptors/InterceptableSuspendMethodTest.kt:[24,23] Unresolved reference: Arc
[ERROR] /Users/manderse/code/quarkusio/quarkus/independent-projects/arc/tests/src/test/kotlin/io/quarkus/arc/test/interceptors/KotlinInterceptorTest.kt:[3,23] Unresolved reference: Arc
[ERROR] /Users/manderse/code/quarkusio/quarkus/independent-projects/arc/tests/src/test/kotlin/io/quarkus/arc/test/interceptors/KotlinInterceptorTest.kt:[24,25] Unresolved reference: Arc
[ERROR] /Users/manderse/code/quarkusio/quarkus/independent-projects/arc/tests/src/test/kotlin/io/quarkus/arc/test/interceptors/NotInterceptedSuspendMethodTest.kt:[3,23] Unresolved reference: Arc
[ERROR] /Users/manderse/code/quarkusio/quarkus/independent-projects/arc/tests/src/test/kotlin/io/quarkus/arc/test/interceptors/NotInterceptedSuspendMethodTest.kt:[18,23] Unresolved reference: Arc
[ERROR] -> [Help 1]
[ERROR
```

don't think its this pr fault but just t ogive context why didn'tconfirm the pr yet.

**Comment by maxandersen** [comment]
okey so I managed to build this when disabling cache (-Dno-build-cache=false) but i still see the issue of the line staying around.
It works just fine when I use quarkus 3.12.

so something in this PR or branch is triggering bad behaviour.

I tried using iterm2 and terminal - same result. works in 3.12 but with this PR it becomes broken.

**Comment by maxandersen** [comment]
also tried plain bash (so none of my zsh customization should influence it). same problem. the prompt keeps "hanging" and even though I can do things like 's' for restart or 'w' for open browser - the terminal is not showing I can do that.

**Comment by MaciejDromin** [comment]
I have just retested this, and for me still works as it is supposed to even on the different OS.
But let's start from the beginning. First I rebased my changed on the top of main to have the latest and greatest and it went without any issues.
Next I build the project executing `./mvnw -Dquickly`
Moving on to testing, I have created a new project by executing `quarkus create app -x picocli whoops` and updating the pom with following values:
```
<quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
<quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
<quarkus.platform.version>999-SNAPSHOT</quarkus.platform.version>
<quarkus-plugin.version>999-SNAPSHOT</quarkus-plugin.version>
```
And now, using the just build CLI jar, I run this project in two different terminals (emulated by Intellij and system) and here are the results:

https://github.com/user-attachments/assets/afa729a4-a805-429f-bf3f-a3c045d591f4

@gsmet @maxandersen I'm not sure why it is happening to You, but like I said, I had no issues with building and running this change, and I tested this on different OS and shells. Maybe there are some steps that You do differently?

**Comment by cescoffier** [comment]
@maxandersen What should we do about this one?

**Comment by cescoffier** [comment]
@maxandersen What should we do about this one?

**Comment by maxandersen** [comment]
@MaciejDromin sorry for dropping ball on this one - it still fails for me so not sure how to fix it :/ maybe a shared screen/debug session would help? (sometime in the new year)

**Comment by MaciejDromin** [comment]
@maxandersen Sure, sounds good. Maybe together we will be able to figure out what is happening :D
I don't have any plans yet for January so I'm up for any date, maybe except of 6th

**Comment by cescoffier** [comment]
@maxandersen sounds like you will be able to debug it :-)

**Comment by cescoffier** [comment]
@maxandersen Ping?

**Comment by cescoffier** [comment]
@maxandersen please have a look
